Slim/Plugin/Podcast/ProtocolHandler.pm

```diff
@@ -0,0 +1,91 @@
+package Slim::Plugin::Podcast::ProtocolHandler;
+
+# Logitech Media Server Copyright 2005-2020 Logitech.
+
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License, 
+# version 2.
+
+use base qw(Slim::Player::Protocols::HTTPS);
+
+use strict;
+
+use Slim::Utils::Log;
+use Slim::Utils::Prefs;
+use Slim::Utils::Cache;
+
+my $log   = logger('plugin.podcast');
+my $prefs = preferences('plugin.podcast');
+my $cache = Slim::Utils::Cache->new;
+my $cachePrefix;
+
+Slim::Player::ProtocolHandlers->registerHandler('podcast', __PACKAGE__);
+
+# remove podcast:// protocol to scan real url
+sub scanUrl {
+	my ( $class, $url, $args ) = @_;
+	my $track = $args->{song}->track;
+	my ($scanUrl, $from) = Slim::Plugin::Podcast::Plugin::unwrapUrl($url);	
+	
+	main::INFOLOG && $log->info("Scanning podcast $url for title ", $track->title);
+
+	# as for redirect, need to port to new url the cover/icon set in XMLBrowser
+	if ( my $icon = $cache->get($cachePrefix . $url) ) {
+		$cache->set($cachePrefix . $scanUrl, $icon, '30 days');
+	}
+	
+	$args->{song}->seekdata( { timeOffset => $from } ) if $from;
+	$class->SUPER::scanUrl($scanUrl, $args);
+}
+
+# we want the image to be cached for us (we will port it only when needed)
+sub cacheImage {
+	my ($class, $url, $image, $prefix) = @_;
+	return $cachePrefix = $prefix;
+}

```

Comment:
I'm not sure I understand that this is supposed to do. This would always return true, wouldn't it?

Or is this just a shortcut to populate the `$cachePrefs` value while returning true at the same time? I think that prefix is hard-coded all over the place already. Do we really need to make this a variable here? I know, it's not best practice... but this call here was really confusing me, as it's assigning a static value to a variable to return a truthy value.

Reply:
Here yes it returns true but I made that a general function. In other words, if you return true it means that you have taken care of it. If you return false, it means that you ask LMS to cache for you. I can, per your previous comment, reverse the logic and call it "shouldCacheImage" but I'm not sure it is substantially different. Let me know